jobsdb/jobsdb_test.go

```diff
@@ -455,6 +482,300 @@ var dsListInMemory = []dataSetT{
 	d2,
 }
 
+func setTestEnvs(t *testing.T) {
+	t.Setenv("RSERVER_JOBS_DB_BACKUP_ENABLED", "true")
+	t.Setenv("JOBS_BACKUP_STORAGE_PROVIDER", "MINIO")
+	t.Setenv("RSERVER_JOBS_DB_BACKUP_BATCH_RT_ENABLED", "true")
+	t.Setenv("JOBS_BACKUP_BUCKET", "backup-test")
+	t.Setenv("RUDDER_TMPDIR", "/tmp")
+	t.Setenv(config.TransformKey("JobsDB.maxDSSize"), "10")
+	t.Setenv(config.TransformKey("JobsDB.migrateDSLoopSleepDuration"), "3")
+	t.Setenv("JOBS_BACKUP_BUCKET", bucket)
+	t.Setenv("JOBS_BACKUP_PREFIX", prefix)
+	t.Setenv("MINIO_ACCESS_KEY_ID", accessKeyId)
+	t.Setenv("MINIO_SECRET_ACCESS_KEY", secretAccessKey)
+	t.Setenv("MINIO_ENDPOINT", minioEndpoint)
+	t.Setenv("MINIO_SSL", "false")
+}
+
+func TestBackupTable(t *testing.T) {
+	// running minio container on docker
+	pool, err := dockertest.NewPool("")
+	if err != nil {
+		log.Fatalf("Could not connect to docker: %s", err)
+	}
+
+	minioResource, err := pool.RunWithOptions(&dockertest.RunOptions{
+		Repository: "minio/minio",
+		Tag:        "latest",
+		Cmd:        []string{"server", "/data"},
+		Env: []string{
+			fmt.Sprintf("MINIO_ACCESS_KEY=%s", accessKeyId),
+			fmt.Sprintf("MINIO_SECRET_KEY=%s", secretAccessKey),
+			fmt.Sprintf("MINIO_SITE_REGION=%s", region),
+		},
+	})
+	if err != nil {
+		panic(fmt.Errorf("Could not start resource: %s", err))
+	}
+	defer func() {
+		if err := pool.Purge(minioResource); err != nil {
+			log.Printf("Could not purge resource: %s \n", err)
+		}
+	}()
+
+	minioEndpoint = fmt.Sprintf("localhost:%s", minioResource.GetPort("9000/tcp"))
+
+	// check if minio server is up & running.
+	if err := pool.Retry(func() error {
+		url := fmt.Sprintf("http://%s/minio/health/live", minioEndpoint)
+		resp, err := http.Get(url)
+		if err != nil {
+			return err
+		}
+		if resp.StatusCode != http.StatusOK {
+			return fmt.Errorf("status code not OK")
+		}
+		return nil
+	}); err != nil {
+		log.Fatalf("Could not connect to docker: %s", err)
+	}
+	fmt.Println("minio is up & running properly")
```

Comment:
please replace all occurences of `fmt.Println` with `t.Log` statements

Reply:
done